File: 005-一个复杂动画的实现_OC/005-一个复杂动画的实现_OC/ViewController.m
//
//  ViewController.m
//  005-一个复杂动画的实现_OC
//
//  Created by 马遥 on 2018/1/23.
//  Copyright © 2018年 马遥. All rights reserved.
//

#import "ViewController.h"
#import "AnimationView.h"

@interface ViewController ()
@property (nonatomic, strong) AnimationView *animationView;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    _animationView = [[AnimationView alloc] initWithFrame:self.view.bounds];
    _animationView.animationCompleted = ^(BOOL completed) {
        NSLog(@"动画已结束 ： %@",completed?@"YES":@"NO");
    };
    
    UITapGestureRecognizer *tap = [[UITapGestureRecognizer alloc] initWithTarget:self action:@selector(startAnimation:)];
    [self.view addGestureRecognizer:tap];
    [self.view addSubview:_animationView];
}

- (void)viewDidAppear:(BOOL)animated {
    [super viewDidAppear:animated];
    [self startAnimation:nil];
}

- (void)startAnimation:(UITapGestureRecognizer *)tap {
    NSLog(@"点击=====开始动画");
    [_animationView startAnimation];
    
}


@end

File: 004-一个复杂动画的实现/004-一个复杂动画的实现/ViewController.swift
//
//  ViewController.swift
//  004-一个复杂动画的实现
//
//  Created by 马遥 on 2018/1/23.
//  Copyright © 2018年 马遥. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    let animationView: AnimationView = AnimationView.init(frame: UIScreen.main.bounds)
    
    
    override func viewDidLoad() {
        super.viewDidLoad()
        let tap = UITapGestureRecognizer.init(target: self, action: #selector(beginAnimation(tap:)))
        
        self.view.addGestureRecognizer(tap)
    }


    @objc func beginAnimation(tap:UITapGestureRecognizer) {
        if self.view.subviews.contains(animationView) {
            animationView.removeFromSuperview()
        }
        self.view.addSubview(animationView)
        animationView.startAnimation()
    }
}

